### ISSUE TITLE
Twitch download plugin should use new Helix API

### ISSUE THREAD

**Issue by qskwood** [qskwood/stream-archiver#2]

**Comment by qskwood** [comment]
The new API requires expiring OAuth tokens.

**Comment by qskwood** [comment]
To get an OAuth token use

```bash
curl -s -X POST "https://id.twitch.tv/oauth2/token?client_id=${clientid}&client_secret=${clientsecret}&grant_type=client_credentials" | jq -r '.access_token'
```

and to get the channel information use

```bash
curl -s -H "Client-ID: ${clientid}" -H "Authorization: Bearer ${bearertoken}" -X GET "https://api.twitch.tv/helix/channels/?broadcaster_id=${channelid}"
```

It would be polite to only grab a new OAuth token when necessary rather than every time the script is run.

**Comment by qskwood** [comment]
To get the current expiration status of a token use

```bash
 curl -s -H "Authorization: OAuth ${bearertoken}" https://id.twitch.tv/oauth2/validate | jq '.expires_in'
```

If it returns `null` it is invalid and probably expired.

**Comment by qskwood** [comment]
> and to get the channel information use
> 
> ```shell
> curl -s -H "Client-ID: ${clientid}" -H "Authorization: Bearer ${bearertoken}" -X GET "https://api.twitch.tv/helix/channels/?broadcaster_id=${channelid}"
> ```

To get whether or not the channel is live a different query is necessary. To get the channel's status use

```bash
curl -s -H "Client-ID: ${clientid}" -H "Authorization: Bearer ${bearertoken}" -X GET "https://api.twitch.tv/helix/streams/?user_login=${channelname}" | jq '.data[0].id'
```

If it returns `null` a live stream wasn't found, which probably means the channel is offline.

As a benefit looking up the channel ID is not necessary.

**Comment by qskwood** [comment]
The client secret can be stored in `./etc/stream-archiver/apis.json'. The OAuth bearer token can be stored in `./var/spool/stream-archiver/`.

The files storing these values should have permissions `0600`.

**Comment by qskwood** [comment]
Implemented and testing in #4

**Comment by qskwood** [comment]
PR merged.
